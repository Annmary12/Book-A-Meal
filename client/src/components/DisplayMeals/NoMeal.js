import React from 'react';
import { Link } from 'react-router-dom';
import steamingFood from '../../../assets/img/steaming.svg';

const NoMeal = () => (
  <div style={{ textAlign: 'center' }}>
    <img src={steamingFood} alt="No meal" /><br />
    No menu has been set for the day.<br />
    Our caterers are preparing something amazing!.<br /><br />
    <Link to="/signup">Sign up</Link> to be notified when menus are set
  </div>
);

export default NoMeal;
