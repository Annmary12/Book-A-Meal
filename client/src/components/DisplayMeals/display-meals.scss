.meals {
  margin-top: 27px;

  @media only screen and (max-width: 600px) {
    margin-top: 0px;
    display: flex;
		flex-direction: column;
  }

  @media only screen and (min-width: 600px) {
    display: grid;
		grid-template-columns: repeat(2,1fr);
  }

  @media only screen and (min-width: 768px) {
    display: grid;
		grid-template-columns: repeat(3,1fr);
  }

  @media only screen and (min-width: 992px) {
    display: grid;
		grid-template-columns: repeat(4,1fr);
  }
}