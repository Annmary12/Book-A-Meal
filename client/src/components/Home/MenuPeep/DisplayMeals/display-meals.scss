@import '../../../../../assets/scss/_variables';

.meals {
  margin-top: 27px;

  @media only screen and (max-width: 600px) {
    margin-top: 0px;
    display: flex;
		flex-direction: column;
  }

  @media only screen and (min-width: 600px) {
    display: grid;
		grid-template-columns: repeat(2,1fr);
  }

  @media only screen and (min-width: 768px) {
    display: grid;
		grid-template-columns: repeat(3,1fr);
  }

  @media only screen and (min-width: 992px) {
    display: grid;
		grid-template-columns: repeat(4,1fr);
  }
}

.noMenu{
  height: 100%;
  background-color: white;
  width: 92%;
  margin: auto;
  margin-top: 0px;
  box-shadow: $card-shadow;
  border-radius: 5px;
  padding: 15px;
  :global{
    animation: slideInUp .3s ease-in-out;
  }
  @media only screen and (max-width: 600px) {

  }

  @media only screen and (min-width: 1200px) {
    width: 70%;
  }
}