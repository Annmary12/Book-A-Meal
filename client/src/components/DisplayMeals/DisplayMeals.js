import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';
import shortid from 'shortid';
import MealCard from '../MealCard/MealCard.js';
import NoMeal from './NoMeal.js';
import LoadMoreButton from './LoadMore/LoadMoreButton.js';
import animateWindowScrollBy from '../../utils/animateWindowScrollBy';
import styles from './display-meals.scss';

/**
 * @class DisplayMeals
 */
class DisplayMeals extends Component {
  static propTypes = {
    meals: PropTypes.arrayOf(PropTypes.object).isRequired
  };
  state = {
    visibleMeals: 4
  }

  /**
   * Displays more meals
   *
   * @returns {undefined}
   */
  displayMore = () => {
    this.setState({
      visibleMeals: this.state.visibleMeals + 4
    });
    animateWindowScrollBy(30, 20);
  }

  /**
   * @returns {JSX} - React JSX
   */
  render() {
    const { meals } = this.props;
    const { visibleMeals } = this.state;

    if (meals.length > 0) {
      return (
        <Fragment>
          <div className={styles.meals}>
            {meals.slice(0, visibleMeals).map((meal, index) => (
              <MealCard
                key={index} // eslint-disable-line react/no-array-index-key
                meal={meal}
                order
              />))
            }
          </div>
          {
            visibleMeals < meals.length
            &&
            <LoadMoreButton handleClick={this.displayMore} />
          }
        </Fragment>
      );
    }
    return <NoMeal />;
  }
}


export default DisplayMeals;
