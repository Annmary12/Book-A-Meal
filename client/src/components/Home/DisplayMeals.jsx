import React from 'react';
import PropTypes from 'prop-types';
import MealCard from './MealCard.jsx';
import NoMeal from './NoMeal.jsx';

const DisplayMeals = (props) => {
  if (props.meals.length > 0) {
    return (
      <div className="meals">
        {props.meals.map(meal => (
          <MealCard
            key={meal.id}
            image={meal.img}
            description={meal.description}
            title={meal.title}
            price={meal.price}
          />))
        }
      </div>
    );
  }
  return <NoMeal />;
};

DisplayMeals.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default DisplayMeals;
