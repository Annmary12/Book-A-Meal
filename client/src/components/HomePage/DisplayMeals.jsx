import React, { Component } from 'react';
import PropTypes from 'prop-types';
import NoMeal from './NoMeal.jsx';

const DisplayMeals = (props) => (
  props.meals.length > 0 
    ? <div className="meals">
        {
          props.meals.map(meal => (
            <MealCard 
              key={meal.id} 
              image={meal.img} 
              description={meal.description} 
              title={meal.title} 
              price={meal.price} 
            />
          ))
        }
      </div>
    : <NoMeal /> 
)

DisplayMeals.propTypes = {
  meals: PropTypes.array.isRequired
}

export default DisplayMeals;