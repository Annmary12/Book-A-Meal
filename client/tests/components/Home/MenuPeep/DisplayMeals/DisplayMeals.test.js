import React from 'react';
import DisplayMeals from '../../../../../src/components/Home/MenuPeep/DisplayMeals/DisplayMeals';

describe('DisplayMeals component', () => {
  it('should match snapshot if meals array is empty', () => {
    const wrapper = shallow(
      <DisplayMeals
        meals={[]}
        loggedIn={true}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if logged is false', () => {
    const meals = [{
      id: 'test-id',
      title: 'Rice and beans'
    }]
    const wrapper = shallow(
      <DisplayMeals
        meals={meals}
        loggedIn={false}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if meals array is not empty', () => {
    const meals = [{
      id: 'test-id',
      title: 'Rice and beans'
    }]
    const wrapper = shallow(
      <DisplayMeals
        meals={meals}
        loggedIn={true}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });


  it('should match snapshot if visibleMeals is less than meal length', () => {
    const meals = [{
      id: 'test-id',
      title: 'Rice and beans'
    }]
    const wrapper = shallow(
      <DisplayMeals
        meals={meals}
        loggedIn={true}
      />
    );
    wrapper.instance().setState({ visibleMeals: 0 });

    expect(wrapper).toMatchSnapshot();
  });

  describe('displayMore method', () => {
    it('should increase visibleMeals in state by 4 when called', () => {
      const wrapper = shallow(
        <DisplayMeals
          meals={[]}
          loggedIn={true}
        />
      );
      const prevState = wrapper.instance().state;
      wrapper.instance().displayMore();
      const newState = wrapper.instance().state;
      expect(newState.visibleMeals).toEqual(prevState.visibleMeals + 4)
    })
  })
});