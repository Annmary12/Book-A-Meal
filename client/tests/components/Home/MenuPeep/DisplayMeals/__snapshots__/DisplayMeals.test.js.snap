// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayMeals component should match snapshot if logged is false 1`] = `
<Fragment>
  <div
    className="meals"
  >
    <MealCard
      action=""
      className=""
      deleteMeal={[Function]}
      key="0"
      meal={
        Object {
          "id": "test-id",
          "title": "Rice and beans",
        }
      }
    />
  </div>
</Fragment>
`;

exports[`DisplayMeals component should match snapshot if meals array is empty 1`] = `
<div
  className="noMenu"
>
  <NoMenu
    loggedIn={true}
  />
</div>
`;

exports[`DisplayMeals component should match snapshot if meals array is not empty 1`] = `
<Fragment>
  <div
    className="meals"
  >
    <MealCard
      action="order"
      className=""
      deleteMeal={[Function]}
      key="0"
      meal={
        Object {
          "id": "test-id",
          "title": "Rice and beans",
        }
      }
    />
  </div>
</Fragment>
`;

exports[`DisplayMeals component should match snapshot if visibleMeals is less than meal length 1`] = `
<Fragment>
  <div
    className="meals"
  />
  <LoadMore
    handleClick={[Function]}
  />
</Fragment>
`;
